File: data/imported/knowledge/3d/net/merged/snippets/snippet_001.cs
[Fact]
        public void ArrayListAdapterInVertexElementTemplate_ShouldWork()
        {
            // Verify ArrayListAdapter works correctly through VertexElementTemplate
            var element = new TestVertexElementTemplate();
            
            Assert.NotNull(element.Data);
            Assert.Equal(0, element.Data.Count);
            
            element.Data.Add(1.5);
            element.Data.Add(2.5);
            Assert.Equal(2, element.Data.Count);
            
            Assert.Equal(1.5, element.Data[0]);
            Assert.Equal(2.5, element.Data[1]);
            
            element.Data.Clear();
            Assert.Empty(element.Data);
        }

File: data/imported/knowledge/3d/java/merged/snippets/snippet_001.java
@Test
    public void ArrayListAdapterInVertexElementTemplate_ShouldWork() {
        TestVertexElementTemplate element = new TestVertexElementTemplate();

        assertNotNull(element.getData());
        assertEquals(0, element.getData().size());

        element.getData().add(1.5);
        element.getData().add(2.5);
        assertEquals(2, element.getData().size());

        assertEquals(1.5, element.getData().get(0), 1e-10);
        assertEquals(2.5, element.getData().get(1), 1e-10);

        element.getData().clear();
        assertEquals(0, element.getData().size());
    }